mitosheet/src/components/taskpanes/Graph/GraphStyleTab.tsx

```diff
@@ -291,6 +295,576 @@ function GraphStyleTab(props: {
+                                        const graphParamsCopy: GraphParams = JSON.parse(JSON.stringify(prevGraphParams)); 
+                                        return {
+                                            ...graphParamsCopy,
+                                            graphStyling: {
+                                                ...graphParamsCopy.graphStyling,
+                                                legend: {
+                                                    ...graphParamsCopy.graphStyling.legend,
+                                                    x: (newX as number | undefined)
+                                                }
+                                            } 
+                                        }
+                                    })
+                                    props.setGraphUpdatedNumber(old => old + 1)
+                                }}     
+                            />
+                        </Row>
+                        <Row justify='space-between' align='center' title='The y position of the legend'>
+                            <Col>
+                                <p>
+                                    Y position
+                                </p>
+                            </Col>
+                            <Input 
+                                value={graphStylingParams.legend.y ? graphStylingParams.legend.y.toString() : ''} 
+                                type='number'
+                                width='small'
+                                // Set default according the https://plotly.com/python/reference/layout/#layout-legend-y
+                                placeholder={graphStylingParams.legend.orientation === 'v' ? '1.00' : graphStylingParams.xaxis.rangeslider.visible ? '1.10' : "-0.10"}
+                                onChange={(e) => {
+                                    const newY = e.target.value === '' ? undefined : e.target.value
+                                    props.setGraphParams(prevGraphParams => {
+                                        const graphParamsCopy: GraphParams = JSON.parse(JSON.stringify(prevGraphParams)); 
+                                        return {
+                                            ...graphParamsCopy,
+                                            graphStyling: {
+                                                ...graphParamsCopy.graphStyling,
+                                                legend: {
+                                                    ...graphParamsCopy.graphStyling.legend,
+                                                    y: (newY as number | undefined)
+                                                }
+                                            } 
+                                        }
+                                    })
+                                    props.setGraphUpdatedNumber(old => old + 1)
```

Comment:
I wanna see if I can think of a new helper function that makes this cleaner (the nesting is just brutal). Will investigate when I do a full review!